static/app/components/timeline/index.tsx

```diff
@@ -0,0 +1,177 @@
+  title,
+  children,
+  icon,
+  timeString,
+  startTimeString,
+  colorConfig = {primary: 'gray300', secondary: 'gray200'},
+  isActive = false,
+  onClick,
+  onMouseEnter,
+  onMouseLeave,
+}: ItemProps) {
+  const hasRelativeTime = defined(startTimeString);
+  const placeholderTime = useMemo(() => new Date().toTimeString(), []);
+  const {
+    displayTime,
+    date,
+    timeWithMilliseconds: preciseTime,
+  } = hasRelativeTime
+    ? getFormattedTimestamp(timeString, startTimeString, true)
+    : getFormattedTimestamp(timeString, placeholderTime);
+
+  return (
+    <Row
+      color={colorConfig.secondary}
+      hasLowerBorder={isActive ?? false}
+      onClick={onClick}
+      onMouseEnter={onMouseEnter}
+      onMouseLeave={onMouseLeave}
+    >
+      <IconWrapper colorConfig={colorConfig} hasIconBorder={isActive ?? false}>
+        {icon}
+      </IconWrapper>
+      <Title color={colorConfig.primary}>{title}</Title>
+      <Timestamp>
+        <Tooltip title={`${preciseTime} - ${date}`}>{displayTime}</Tooltip>
+      </Timestamp>
+      <Spacer hasLine={isActive ?? false} />
+      <Content>{children}</Content>
+    </Row>
+  );
+}
+
+interface GroupProps {
+  children: React.ReactNode;
+}
+
+export function Group({children}: GroupProps) {
+  return <GroupWrapper>{children}</GroupWrapper>;
+}
+
+const GroupWrapper = styled('div')`
+  position: relative;
+  /* vertical line connecting items */
+  &::before {
+    content: '';
+    position: absolute;
+    left: 10.5px;
+    width: 1px;
+    top: 0;
+    bottom: 0;
+    background: ${p => p.theme.border};
+  }
+`;
+
+const Row = styled('div')<{color: string; hasLowerBorder: boolean}>`
+  position: relative;
+  color: ${p => p.theme.subText};
+  display: grid;
+  align-items: center;
+  grid-template: auto auto / 22px 1fr auto;
+  grid-column-gap: ${space(1)};
+  border-bottom: 1px solid ${p => (p.hasLowerBorder ? p.theme[p.color] : 'transparent')};
+  margin: ${space(1)} 0;
+  &:first-child {
+    margin-top: 0;
+  }
+  &:last-child {
+    margin-bottom: 0;
+    background: ${p => p.theme.background};
+  }
+  &:last-child > :last-child {
+    margin-bottom: ${p => (p.hasLowerBorder ? space(1) : 0)};
+  }
+`;
+
+const IconWrapper = styled('div')<{colorConfig: ColorConfig; hasIconBorder: boolean}>`
+  grid-column: span 1;
+  border-radius: 100%;
+  border: 1px solid;
+  border-color: ${p =>
+    p.hasIconBorder ? p.theme[p.colorConfig.secondary] : 'transparent'};
+  color: ${p => p.theme[p.colorConfig.primary]};
+  background: ${p => p.theme.background};
+  svg {
+    display: block;
+    margin: ${space(0.5)};
+  }
+`;
+
+const Title = styled('p')<{color: string}>`
+  color: ${p => p.theme[p.color]};
+  margin: 0;
+  font-weight: bold;
+  text-transform: capitalize;
+  grid-column: span 1;
+`;
+
+const Timestamp = styled('p')`
+  margin: 0 ${space(1)};
+  color: ${p => p.theme.subText};
+  span {
+    text-decoration: underline dashed ${p => p.theme.subText};
+  }
```

Comment:
with skipWrapper there won't be a child span
```suggestion
  span {
    text-decoration: underline dashed ${p => p.theme.subText};
  }
```